pkg/network/http/ebpf_gotls.go

```diff
@@ -178,62 +209,48 @@ func (p *GoTLSProgram) addInspectionResultToMap(result *bininspect.Result, binPa
 
 func (p *GoTLSProgram) attachHooks(result *bininspect.Result, binPath string) error {
 	uid := getUID(binPath)
-	for i, offset := range result.Functions[bininspect.ReadGoTLSFunc].ReturnLocations {
-		probeID := manager.ProbeIdentificationPair{
-			EBPFSection:  readReturnProbe,
-			EBPFFuncName: readReturnFuncName,
-			UID:          makeReturnUID(uid, i),
-		}
-		err := p.manager.AddHook("", &manager.Probe{
-			ProbeIdentificationPair: probeID,
-			BinaryPath:              binPath,
-			// Each return probe needs to have a unique uid value,
-			// so add the index to the binary UID to make an overall UID.
-			UprobeOffset: offset,
-		})
-		if err != nil {
-			return fmt.Errorf("could not add hook to read return in offset %d due to: %w", offset, err)
+
+	for function, uprobes := range functionToProbes {
+		if functionsConfig[function].IncludeReturnLocations && uprobes.returnInfo != nil {
+			for i, offset := range result.Functions[function].ReturnLocations {
+				returnProbeID := manager.ProbeIdentificationPair{
+					EBPFSection:  uprobes.returnInfo.ebpfSection,
+					EBPFFuncName: uprobes.returnInfo.ebpfFunctionName,
+					UID:          makeReturnUID(uid, i),
```

Comment:
among different return locations of different functions the UID will be the same (the first `return` in write-return and read-return will have the same uid)
do we want to change it?

Reply:
yeah you're right! I'll change it

Reply:
Talked about it but adding here for documentation :) 
The entire ProbeIdentificationPair needs to be unique, so it's okay if two probes has the same uid if the section / funcName is different (and this is the case here is read / write)